### ISSUE TITLE
DQN calc_target_f error in exaalt_develop branch

### ISSUE THREAD

**Issue by armenlanl** [exalearn/EXARL#255]
I've been trying to get the ExaCartPoleStatic-v0 env working with the DQN agent on my local machine. I've setup the conda env to match the requirements found in the setup.py file but am running into an error with the bellman equation calculation function.

When setting the `target_f[0][action_index]` in the code block down below I receiving an error that you cannot assign a value in an EagerTensor

```ruby
        state, action, reward, next_state, done = exp
        np_state = self.flatten_observation(state)
        np_next_state = self.flatten_observation(next_state)
        reward = tf.cast(reward, tf.float32)
        np_state = tf.convert_to_tensor(np_state)
        np_next_state = tf.convert_to_tensor(np_next_state)
        expectedQ = 0
        with tf.device(self.device):
            if not done:
                expectedQ = tf.multiply(self.gamma, tf.reduce_max(self.target_model.predict_on_batch(np_next_state)))
                expectedQ = tf.cast(expectedQ, tf.float32)
            target = tf.add(reward, expectedQ)
            target_f = self.target_model.predict_on_batch(np_state)
        # For handling continuous to discrete actions
        action_idx = action if self.is_discrete else np.where(self.actions == action)[1]

        target_f[0][action_idx] = target
        return target_f[0]
```
This results in the current error show below in this screenshot:
![image](https://github.com/exalearn/EXARL/assets/137205313/73742bbf-0b57-454d-b7a3-c57c5352b1f9)

My current solution to fix this is converting the tensor into a numpy array, setting the value required and converting it back to a tensor. I am currently validating that this is working with convergence checks. 

Could this be an error in my environment? If not the fixed code block can be seen below:

```ruby
        state, action, reward, next_state, done = exp
        np_state = self.flatten_observation(state)
        np_next_state = self.flatten_observation(next_state)
        reward = tf.cast(reward, tf.float32)
        np_state = tf.convert_to_tensor(np_state)
        np_next_state = tf.convert_to_tensor(np_next_state)
        expectedQ = 0
        with tf.device(self.device):
            if not done:
                expectedQ = tf.multiply(self.gamma, tf.reduce_max(self.target_model.predict_on_batch(np_next_state)))
                expectedQ = tf.cast(expectedQ, tf.float32)
            target = tf.add(reward, expectedQ)
            target_f = self.target_model.predict_on_batch(np_state)
        # For handling continuous to discrete actions
        action_idx = action if self.is_discrete else np.where(self.actions == action)[1]

        target_f_np = target_f.numpy()
        target_f_np[0][action_idx] = target
        target_f = tf.convert_to_tensor(target_f_np, dtype=tf.float32)
        return target_f[0]
```

**Comment by armenlanl** [comment]
Here is the reward plot after the change to running with this command:

`python exarl/driver --workflow sync --agent DQN-v0 --env ExaCartPoleStatic-v0 --n_episodes 250 --n_steps 500`

![image](https://github.com/exalearn/EXARL/assets/137205313/8bd6533a-da7d-4f4d-86bf-aea47a74f39d)
